sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -252,6 +268,32 @@ protected static boolean validFieldName(String fieldName) {
         return result;
     }
 
+    /**
+     * Based on the proposed index and the list of existing indexes, this method checks
+     * whether another index can be created.  Currently the limit for TEXT indexes is 1.
+     * JSON indexes are unlimited.
+     *
+     * @param index the proposed index
+     * @param existingIndexes the list of already existing indexes
+     * @return whether the index limit has been reached
+     */
+    @SuppressWarnings("unchecked")
+    protected static boolean indexLimitReached(Index index, Map<String, Object> existingIndexes) {
+        if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+            for (String name : existingIndexes.keySet()) {
+                Map<String, Object> existingIndex = (Map<String, Object>) existingIndexes.get(name);
+                String type = (String) existingIndex.get("type");
+                if (type.equalsIgnoreCase("text") && !name.equalsIgnoreCase(index.indexName)) {
```

Comment:
When you say _existing code_ I am assuming that you mean the code in this PR since prior to this PR index settings did not exist.  In the event that an index already exists with the same name, same fields but different settings, the logic would drop us into the index creation process which would fail via SQLite (desired result) since a table already exists with the same name.  This is pretty much the same process as described in https://github.com/cloudant/sync-android/pull/112#discussion_r28170373